lib/src/utils/hooks.ts

```diff
@@ -25,3 +25,17 @@ export function useInterval(cb: () => void, ms: number, deps: any[] = []) {
     return clearLastInterval;
   }, [ms, ...deps]);
 }
+
+export function useTimeout(cb: () => void, ms: number, deps: any[] = []) {

```

Comment:
remove this function, use useEffect and a debounced function, use useContext to store this function over rerenders.